clusterloader2/pkg/framework/client/objects.go

```diff
@@ -99,6 +99,24 @@ func DeleteNamespace(c clientset.Interface, namespace string) error {
 	return RetryWithExponentialBackOff(retryFunction(deleteFunc, apierrs.IsNotFound))
 }
 
+// WaitForDeleteNamespace waits untils namespace is terminated.
+func WaitForDeleteNamespace(c clientset.Interface, namespace string) error {
+	retryWaitFunc := func() (bool, error) {
+		_, innerErr := c.CoreV1().Namespaces().Get(namespace, metav1.GetOptions{})
+		if innerErr != nil {
+			if apierrs.IsNotFound(innerErr) {
+				return true, nil
+			}
+			if IsRetryableAPIError(innerErr) {
+				return false, nil
+			}
+			return false, innerErr
+		}
+		return false, nil
+	}
+	return wait.PollImmediateInfinite(5*time.Second, retryWaitFunc)
```

Comment:
You shouldn't be waiting infinitely for this. There are some known bugs iirc which can lead to namespace being in "Terminating" state forever (due to some non-deletable resources).

Reply:
Timeout added.